pkg/operator/configobserver/node/observe_latency_profile.go

```diff
@@ -0,0 +1,207 @@
+		return existingConfig, errs
+	}
+
+	// get the arg val pairs for currently set latency profile
+	// and set observed config
+	observedConfig := make(map[string]interface{})
+	for _, latencyConfig := range l.latencyConfigs {
+		profileValue, ok := latencyConfig.ProfileConfigValues[configNode.Spec.WorkerLatencyProfile]
+		if !ok {
+			// log that this is an unsupported latency profile
+			// Note: In case new latency profiles are added in the future in openshift/api
+			// this could break cluster upgrades although we're not passing an error here
+			// to ensure that ConfigObservationController doesn't reach degraded state.
+			klog.Warningf("unsupported worker latency profile found in nodes.config.openshift.io/cluster Spec.WorkerLatencyProfile = %v", configNode.Spec.WorkerLatencyProfile)
+			return existingConfig, errs
+		}
+
+		err = unstructured.SetNestedStringSlice(observedConfig, []string{profileValue}, latencyConfig.ConfigPath...)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+	// in case any error(s) were encountered
+	if len(errs) > 0 {
+		return existingConfig, errs
+	}
+
+	return observedConfig, errs
+}
+
+type revisionDiffProfileSuppressor struct {
+	operatorClient  v1helpers.StaticPodOperatorClient
+	configMapLister listersv1.ConfigMapNamespaceLister
+	usedConfigPaths [][]string
+}
+
+// NewSuppressConfigUpdateUntilSameProfileFunc is used to create a conditional func (shouldSuppressConfigUpdatesFunc)
+// that can be used by the latency profile config observer to determine if a new update to observedConfig should be rolled out or not.
+// It uses a config map lister and status obtained from a static pod operator client to check if all active revisions on the cluster have
+// common values for the required latency config paths or not. Config maps generated by installer controller
+// are read in order to determine the current active static pod revision and compare observedConfig values from it.
+func NewSuppressConfigUpdateUntilSameProfileFunc(
+	operatorClient v1helpers.StaticPodOperatorClient,
+	configMapLister listersv1.ConfigMapNamespaceLister,
+	latencyConfigs []LatencyConfigProfileTuple,
+) (f func() (bool, error)) {
+
+	usedConfigPaths := make([][]string, len(latencyConfigs))
+	for i, latencyConfig := range latencyConfigs {
+		usedConfigPaths[i] = latencyConfig.ConfigPath
+	}
+
+	ret := revisionDiffProfileSuppressor{
+		operatorClient:  operatorClient,
+		configMapLister: configMapLister,
+		usedConfigPaths: usedConfigPaths,
+	}
+
+	// creates an actual conditional fn that can be invoked each time config observer
+	// intends to update observed config
+	return ret.shouldSuppressConfigUpdates
+}
+
+func (s *revisionDiffProfileSuppressor) shouldSuppressConfigUpdates() (suppress bool, err error) {
+	_, operatorStatus, _, err := s.operatorClient.GetStaticPodOperatorState()
+	if err != nil {
+		return false, err
+	}
+
+	// Collect the unique set of revisions for all nodes in cluster
+	revisionMap := map[int32]struct{}{}
```

Comment:
why `struct{}`? , `bool` is simpler to use. You can also use `sets` if you want.

Reply:
Changed it to simpler `bool`.

Reply:
Also used similar `bool` in 
```
revisionMap := make(map[int32]bool)
```
in controller.go line 114.